tests/unit/spec/base/mozilla-lazy-load.js

```diff
@@ -0,0 +1,150 @@
+            $target.appendTo($('body'));
+        });
+
+        afterEach(function() {
+            $target.remove();
+        });
+
+        it('should observe elements as expected', function() {
+            spyOn(Mozilla.LazyLoad, 'registerObserver').and.returnValue({
+                observe: sinon.spy()
+            });
+            var observer = Mozilla.LazyLoad.observe('.observer-list-test img');
+            expect(observer.observe.calledTwice).toBeTruthy();
+        });
+    });
+
+    describe('Mozilla.LazyLoad.observerCallback', function() {
+
+        var changes = [
+            {
+                intersectionRatio: 0,
+                target: {
+                    src: '/foo/placeholder.png',
+                    dataset: {
+                        src: '/foo/image1.png'
+                    },
+                    onload: function() {}
+                }
+            },
+            {
+                intersectionRatio: 0.1,
+                target: {
+                    src: '/foo/placeholder.png',
+                    dataset: {
+                        src: '/foo/image2.png'
+                    },
+                    onload: function() {}
+                }
+            },
+            {
+                intersectionRatio: 0,
+                target: {
+                    src: '/foo/placeholder.png',
+                    dataset: {
+                        src: '/foo/image3.png'
+                    },
+                    onload: function() {}
+                }
+            }
+        ];
+
+        var observer = { unobserve: function() {} };
+
+        it('should lazy load images only when they intersect', function() {
+            spyOn(observer, 'unobserve');
+            Mozilla.LazyLoad.observerCallback(changes, observer);
+            expect(changes[0].target.src).toEqual('/foo/placeholder.png');
+            expect(changes[1].target.src).toEqual('/foo/image2.png');
+            expect(changes[2].target.src).toEqual('/foo/placeholder.png');
+            expect(observer.unobserve).toHaveBeenCalledWith(changes[1].target);
+        });
+    });
+
+    describe('Mozilla.LazyLoad.loadAll', function() {
+
+        var $target;
+
+        beforeEach(function () {
+            var tpl = '<div class="observer-list-test">' +
+                        '<img class="image1" src="/img/placeholder.png" data-src="/img/foo.png">' +
+                        '<img class="image2" src="/img/placeholder.png" data-src="/img/foo.png">' +
+                      '</div>';
+            $target = $(tpl);
+            $target.appendTo($('body'));
+        });
+
+        afterEach(function(){
+            $target.remove();
+        });
+
+        it('should register an IntersectionObserver correctly', function() {
+            Mozilla.LazyLoad.loadAll('.observer-list-test img');
+            expect($target.find('.image1').attr('src')).toEqual('/img/foo.png');
+            expect($target.find('.image2').attr('src')).toEqual('/img/foo.png');
```

Comment:
Do you think it's worth testing to make sure `data-src` is removed after updating `src`?